File: App/Models/HiveOS/HiveOSWallet.cs
using Newtonsoft.Json;

namespace GoblinFarm.Models.HiveOS
{
    public class HiveOSWallet
    {
        [JsonProperty("id")]
        public int Id { get; set; }

        [JsonProperty("name")]
        public string Name { get; set; }

        [JsonProperty("coin")]
        public string Coin { get; set; }

        [JsonProperty("wal")]
        public string Wal { get; set; }

        [JsonProperty("source")]
        public string Source { get; set; }

        [JsonProperty("balance")]
        public WalletBalance Balance { get; set; }

        public class WalletBalance
        {
            [JsonProperty("description")]
            public string Description { get; set; }

            [JsonProperty("value")]
            public decimal Value { get; set; }

            [JsonProperty("value_fiat")]
            public decimal ValueFiat { get; set; }

        }
    }
}

File: App/ClientApp/src/models/hive-os/HiveOSWallet.ts
export interface HiveOSWallet {
  id: number;
  name: string;
  wal: string;
  source: string;
  balance: HiveOSWalletBalance;
}

export interface HiveOSWalletBalance {
  description: string;
  value: number;
  valueFiat: number;
}
